demos/ret2spec_common.h

```diff
@@ -0,0 +1,21 @@
+// Recursion depth should be equal or greater than the RSB size, but not
+// excessively high because of the possibility of stack overflow.
+constexpr size_t kRecursionDepth = 64;
+
+// Modular function pointers that provide different functionality in the
+// same-address-space and cross-address-space version. We use this design,
+// because the same-address-space version works everywhere including on systems
+// that do not allow forced pinning of processes to cores (e.g. MacOS), the
+// cross-address-space version is Linux only.
+extern void (*conditionally_unschedule)();
+extern void (*unschedule_or_start_returns_false)();
+
+// Global variables used to avoid passing parameters through recursive function
+// calls. Since we flush whole stack frames from the cache, it is important not
+// to store on stack any data that might be affected by being flushed from
+// cache.
+extern size_t current_offset;
+extern const std::array<BigByte, 256> *oracle_ptr;
```

Comment:
Suggest making these arguments of `Ret2specLeakByte` so these variables can be file-local rather than entirely global.

Reply:
That breaks the attack because it stores them to the stack frame which is evicted from the cache.